ui/pages/swaps/view-quote/view-quote.js

```diff
@@ -460,7 +462,7 @@ export default function ViewQuote() {
     : null;
 
   let ethBalanceNeededStx;
-  if (smartTransactionsError?.balanceNeededWei) {
+  if (isSmartTransaction && smartTransactionsError?.balanceNeededWei) {
```

Comment:
I'm not intimately familiar with this logic — is the issue that when `isSmartTransaction` flips from `true` to `false`, `smartTransactionsError` could still have something in it? If so, then would it make sense to reset `smartTransactionsError`?